woocommerce/woocommerce-ios
# Order details: add-ons from `_pao_ids` metadata - storage layer changes

Part of #10419 


## Why


Currently, we're getting the product add-ons of an order line item by manually mapping the metadata entries with the product and global add-ons that need to be synced separately. Starting Product Add-Ons v5.0.2 pe5pgL-3wp-p2#comment-2967, a new metadata field `_pao_ids` became available to get a list of add-ons of an order line item. The add-ons currently share the same type `OrderItemAttribute` as other order item meta attributes like product variations. As we want to differentiate the add-ons from other order item attributes like for a https://github.com/woocommerce/woocommerce-ios/issues/10419, this PR adds a new Core Data entity `OrderItemProductAddOn`.

## How

The actual schema changes are in https://github.com/woocommerce/woocommerce-ios/pull/10589/commits/503e277f89254349b2ed5a3506ec822df2db2985.

Please note that the add-on ID is optional because this field isn't available because some plugins like WCPay can remove it while filtering the metadata (pe5pgL-3wp-p2#caveat-2-wcpay-plugin-removes-the-add-on-id). Because `@NSManaged` properties need to be compatible with Objective-C, the `addOnID` has type `NSNumber?` to support nullability as `Int64?` isn't allowed here.

## Testing instructions


Feel free to do a confidence check by first running the app in the logged-in state from another branch, then launching from the PR branch to make sure the app works as before.


---
